homeassistant/components/camera/synology.py

```diff
@@ -0,0 +1,104 @@
+"""
+Support for IP Cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.generic/
+"""
+import logging
+
+import json
+from json import loads
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+from homeassistant.components.camera import DOMAIN, Camera
+from homeassistant.helpers import validate_config
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Setup a Synology IP Camera."""
+    if not validate_config({DOMAIN: config}, {DOMAIN: ['synology_url', 'username', 'password', 'camera_name']},
+                           _LOGGER):
+        return None
+
+    add_devices_callback([SynologyCamera(config)])
+
+# pylint: disable=too-many-instance-attributes
+class SynologyCamera(Camera):
+    """An implementation of a Synology NAS based IP camera."""
+
+    def __init__(self, device_info):
+        """Initialize a Synology Surveillance Station camera."""
+        super().__init__()
+        self._name = device_info.get('name', 'Synology Camera')
+        self._username = device_info.get('username')
+        self._password = device_info.get('password')
+        self._synology_url = device_info['synology_url']
+        self._camera_name = device_info['camera_name']
+        self._stream_id = device_info['stream_id']
+
+        """We need to get a session id to retrieve the still image path and the images themselves"""
+        login_url = self._synology_url + '/webapi/auth.cgi?api=SYNO.API.Auth&method=Login&version=2&account=' + self._username + '&passwd=' + self._password + '&session=SurveillanceStation&format=sid'

```

Comment:
The requests module offers support to work with [parameters](http://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls).